app/src/main/java/fi/metropolia/movesense/component/MovesenseGauge.kt

```diff
@@ -1,24 +1,35 @@
 package fi.metropolia.movesense.component
 
 import android.content.Context
+import androidx.compose.foundation.background
 import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
+import androidx.compose.material3.contentColorFor
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.livedata.observeAsState
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.saveable.rememberSaveable
 import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
 import androidx.compose.ui.viewinterop.AndroidView
 import com.ekn.gruzer.gaugelibrary.MultiGauge
 import com.ekn.gruzer.gaugelibrary.Range
+import fi.metropolia.movesense.type.MeasureType
 import fi.metropolia.movesense.view.measure.MeasureViewModel
 import kotlin.math.PI
 import kotlin.math.atan2
 import kotlin.math.sqrt
+import fi.metropolia.movesense.R
 
 @Composable
 fun MovesenseGauge(measureViewModel: MeasureViewModel) {
```

Comment:
Komponenttien pitäisi olla ViewModelista riippumattomia.

Reply:
Totta. Graafista tämän saman vian korjasin PR #46 . Laitan kohta näitä työn alle.